src/main/java/com/bean00/server/RequestProcessor.java

```diff
@@ -17,27 +17,64 @@ public RequestProcessor(DataStore dataStore) {
     }
 
     public Response processRequest(Request request) throws IOException {
+        String requestMethod = request.getRequestMethod();
+        Response response;
+
+        if (requestMethod.equals(Method.PUT)) {
+            response = putHandler(request);
+        } else if (requestMethod.equals(Method.GET) || requestMethod.equals(Method.HEAD)) {
+            response = getHeadHandler(request);
+        } else {
+            response = methodNotAllowedResponse("GET, HEAD, PUT");
+        }
+
+        return response;
+    }
+
+    private Response putHandler(Request request) throws IOException {
```

Comment:
I would probably change the method name so that it's a verb like `handlePutRequest`

Reply:
Changed